core/iwasm/common/wasm_runtime_common.c

```diff
@@ -3016,6 +3016,50 @@ wasm_runtime_init_wasi(WASMModuleInstanceCommon *module_inst,
         fd_prestats_insert(prestats, dir_list[i], wasm_fd);
     }
 
+    for (i = 0; i < map_dir_count; i++, wasm_fd++) {
+        char mapping_copy[(PATH_MAX * 2) + 3];
+        strncpy(mapping_copy, map_dir_list[i], strlen(map_dir_list[i]));
+        mapping_copy[(PATH_MAX * 2) + 2] = '\0';
+
+        const char *map_mapped = strtok(mapping_copy, "::");
+        const char *map_host = strtok(NULL, "::");
```

Comment:
This PR only handles `wasm_runtime_init_wasi` of libc wasi version, would you also handle the version of libc uvwasi? And it didn't add the related argument for iwasm, would you enable it in this PR or in the future? For example:
```bash
iwasm --map-dir=<guest_dir::host_dir>
```
And we can add the argument multiple times: `--map-dir=<guest_dir1::host_dir1> --map-dir=<guest_dir2::host_dir2>`.

Reply:
I intend to do it similarly to the current `--dir` argument which includes multiple arguments for the same list IIRC.

Reply:
Do you mean somewhat like `iwasm --dir=<dir1,dir2,..>`? No, currently it is `iwasm --dir=<dir1> --dir=<dir2> ...`, like `iwasm --env="key1=value1" --env="key2=value2"`, you can see the help of iwasm. So I think for the map dir, it should be like `iwasm --map-dir=<guest_dir1::host_dir1> --map-dir=<guest_dir2::host_dir2>`?

And could you help change the comment of wasm_runtime_set_wasi_args_ex in wasm_export.h, at least to mention the format of map dir:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/include/wasm_export.h#L425

Reply:
Done.